### PR TITLE
Enable ruff's RET504 rule

### PR DESCRIPTION
**Pull request by marcospri**
See: https://docs.astral.sh/ruff/rules/unnecessary-assign/

I could see this being annoying while debugging etc but it can be cleaned when committing.

### CODE REVIEW

File: lms/services/user.py [hunk 1]
```diff
@@ -126,18 +126,10 @@ def get(self, application_instance, user_id: str) -> User:
 
     def _user_search_query(self, application_instance_id, user_id) -> Select:
         """Generate a query for searching for users."""
-
-        query = select(User)
-
-        # Normally we'd have an `if application_instance_id` here, for a proper
```
Comment (marcospri) [inline lms/services/user.py:132]: I removed all of this. 

Yes, a lot other similar method do build a query conditionally but we can just add that here when we need it I don't think we need to explain it here.

File: lms/views/admin/assignment.py [hunk 2]
```diff
@@ -50,16 +50,15 @@ def assignment_dashboard(self):
             )
         )
 
-        # We can only navigate to assignments via their course, assigmentl.course won't be null
+        # We can only navigate to assignments via their course, assignment.course won't be null
```
Comment (marcospri) [inline lms/views/admin/assignment.py:53]: I should write my own ruff rule about this spelling.

**Review by seanh — APPROVED** [review]
I'm not sure how much I love this linting rule to be honest--there sometimes might be good reason to assign something to a variable right before returning it, either for debugging or just for the readability added by the variable name.

**Comment by marcospri** [comment]
Let's not enable this rule as it can be annoying.

**Pull request closed without merging**